### PR TITLE
Message 2 create scaffolding

### PR DESCRIPTION
**Pull request by rucuriousyet**
Added several crud endpoints etc. Basic API is in place.

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -0,0 +1,47 @@
+hash: f6a96f9503dff0ef5299cedab9c38bc9a0fff93de306eecc90ead499b0e70d05
```
Comment (rucuriousyet) [inline glide.lock]: For those of you who don't know what Glide does, this is a machine generated file. I would ignore it for now.

File: glide.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package: github.com/spot-app/message-api
```
Comment (rucuriousyet) [inline glide.yaml]: Another machine generated file to ignore

File: setup/insomnia.json [hunk 3]
```diff
@@ -0,0 +1,184 @@
+{
```
Comment (rucuriousyet) [inline setup/insomnia.json]: This file is only for use with a REST client, Its also something you can ignore.

File: setup/msg.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (rucuriousyet) [inline setup/msg.json]: Thoughts on this schema?

File: src/architype/event.go [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package architype
+
+// Event is a basic type for serializing JSON
+type Event struct {
```
Comment (rucuriousyet) [inline src/architype/event.go]: For verneMQ

File: setup/msg.json (hunk 4 above)
Comment (wbconnor) [inline setup/msg.json]: Should there be an `attachment` field?

Comment (rucuriousyet) [inline setup/msg.json]: Well my thought process would be that the body should contain all the attachments, whether that be image, audio, video, emoji etc. Eventually we could add a field for replies but its not MVP.

Comment (rucuriousyet) [inline setup/msg.json]: For example, the body could be a subset of markdown for easy formatting and rendering. It has almost everything we'd need.

File: src/handler/handler.go [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	// TODO: We need to authorize
+	sinceStr := c.Query("since")
+	convoID := c.Query("convo")
+	token := c.Request.Header.Get("Forwarded-Authorization")
+
+	tokenItself := util.Token{}
+	err := util.Unmarshal([]byte(token), &tokenItself, "secret")
+	if !util.Check(err, false) {
+		//Bad token
+		c.Status(401)
```
Comment (jacted) [inline src/handler/handler.go]: Missing `return` after `c.Status(401)`

**Comment by rucuriousyet** [comment]
Temporarily closed PR... will reopen when ready to merge.

**Pull request closed without merging**